core/src/main/java/jenkins/mvn/DefaultGlobalSettingsProvider.java

```diff
@@ -20,7 +22,7 @@ public DefaultGlobalSettingsProvider() {
     }
 
     @Override
-    public FilePath supplySettings(AbstractBuild<?, ?> project, TaskListener listener) {
+    public FilePath supplySettings(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull TaskListener listener) {
```

Comment:
Thanks for reviewing @KostyaSha. Do you agree with @oleg-nenashev's statement? Do you agree that this PR does not break binary compatibility as this `FilePath supplySettings(AbstractBuild, TaskListener)` method is still declared in the parent class  GlobalSettingsProvider: 
https://github.com/cyrille-leclerc/jenkins/blob/JENKINS-40665/core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java#L57